internal/cmd/cloud/local_server.go

```diff
@@ -0,0 +1,92 @@
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+)
+
+// TokenResponse represents the token received by the local server's callback handler.
+type TokenResponse struct {
+	Token string
+	State string
+}
+
+// bindLocalServer initializes a LocalServer that will listen on a randomly available TCP port.
+func bindLocalServer() (*localServer, error) {
+	listener, err := net.Listen("tcp4", "127.0.0.1:0")
+	if err != nil {
+		return nil, err
+	}
+
+	return &localServer{
+		listener:   listener,
+		resultChan: make(chan TokenResponse, 1),
+	}, nil
+}
+
+type localServer struct {
+	CallbackPath     string
+	WriteSuccessHTML func(w io.Writer)
+
+	resultChan chan (TokenResponse)
+	listener   net.Listener
+}
+
+func (s *localServer) Port() int {
+	return s.listener.Addr().(*net.TCPAddr).Port
+}
+
+func (s *localServer) Close() error {
+	return safeClose(s.listener)
+}
+
+func (s *localServer) Serve() error {
+	return http.Serve(s.listener, s) //nolint:gosec
+}
+
+func (s *localServer) Wait(ctx context.Context) (TokenResponse, error) {
+	select {
+	case <-ctx.Done():
+		return TokenResponse{}, ctx.Err()
+	case code := <-s.resultChan:
+		return code, nil
+	}
+}
+
+// ServeHTTP implements http.Handler.
+func (s *localServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if s.CallbackPath != "" && r.URL.Path != s.CallbackPath {
```

Comment:
See other comment about setting this parameter in this type.
I think we should set it once in the server (or just consider making this a package constant).
Then I would expect to always expect this path. Otherwise, we could handling requests to `/` as callbacks.

```suggestion
	if r.URL.Path != s.CallbackPath {
```